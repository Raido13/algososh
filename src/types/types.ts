import { Dispatch, SetStateAction } from "react"

export type TFormState = {[key:string]: string};
export type TSetActive = Dispatch<SetStateAction<boolean>>;
export type TSetFormState = Dispatch<SetStateAction<{[key:string]: string}>>;