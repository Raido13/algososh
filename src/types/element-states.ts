export enum ElementStates {
  Default = "default",
  Changing = "changing",
  Modified = "modified"
}

export enum LoaderStates {
  Add = 'add',
  Remove = 'remove',
  Reset = 'reset',
  Ascending = 'ascending',
  Descending = 'descending',
  AddHead = 'addHead',
  AddTail = 'addTail',
  RemoveHead = 'removeHead',
  RemoveTail = 'removeTail',
  AddIdx = 'addIdx',
  RemoveIdx = 'removeIdx'
}

export enum BorderStates {
  Default = '4px solid rgb(0, 50, 255)',
  Changing = '4px solid rgb(210, 82, 255)',
  Modified = '4px solid rgb(127, 224, 81)'
}