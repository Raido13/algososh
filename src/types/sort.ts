export enum Sort {
  Bubble = 'bubble',
  Choise = 'Choise'
}