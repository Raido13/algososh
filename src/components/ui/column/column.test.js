import { ElementStates } from "../../../types/element-states";
import { Column } from "./column";
import { render } from '@testing-library/react';

describe('Компонент Column', () => {  
  test('Отрисовка круга: С индексом', () => {
    const {asFragment} = render(<Column index={'100'} />)
    expect(asFragment().firstChild).toMatchSnapshot();    
  });

  test('Отрисовка круга: В дефолтном состоянии', () => {
    const {asFragment} = render(<Column state={ElementStates.Default} />)
    expect(asFragment().firstChild).toMatchSnapshot();    
  });

  test('Отрисовка круга: В изменяющемся состоянии', () => {
    const {asFragment} = render(<Column state={ElementStates.Changing} />)
    expect(asFragment().firstChild).toMatchSnapshot();    
  });

  test('Отрисовка круга: В модифицированном состоянии', () => {
    const {asFragment} = render(<Column state={ElementStates.Modified} />)
    expect(asFragment().firstChild).toMatchSnapshot();    
  });
})