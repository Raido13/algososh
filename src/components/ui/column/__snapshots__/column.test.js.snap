// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Компонент Column Отрисовка круга: В дефолтном состоянии 1`] = `
<div
  class="content "
>
  <div
    class="column default"
    style="height: 1px;"
  />
  <p
    class="text text_type_column text_color_input mt-3"
  />
</div>
`;

exports[`Компонент Column Отрисовка круга: В изменяющемся состоянии 1`] = `
<div
  class="content "
>
  <div
    class="column changing"
    style="height: 1px;"
  />
  <p
    class="text text_type_column text_color_input mt-3"
  />
</div>
`;

exports[`Компонент Column Отрисовка круга: В модифицированном состоянии 1`] = `
<div
  class="content "
>
  <div
    class="column modified"
    style="height: 1px;"
  />
  <p
    class="text text_type_column text_color_input mt-3"
  />
</div>
`;

exports[`Компонент Column Отрисовка круга: С индексом 1`] = `
<div
  class="content "
>
  <div
    class="column default"
    style="height: 320px;"
  />
  <p
    class="text text_type_column text_color_input mt-3"
  >
    100
  </p>
</div>
`;
